src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -721,77 +639,35 @@ export class MeetingManager implements AudioVideoObserver {
     }
   };
 
-  subscribeToSelectedVideoInputTransformDevice = (
-    callback: (device: Device | VideoTransformDevice) => void
-  ): void => {
-    this.selectedVideoInputTransformDeviceObservers.push(callback);
-  };
-
-  unsubscribeFromSelectedVideoInputTranformDevice = (
-    callbackToRemove: (device: Device | VideoTransformDevice) => void
-  ): void => {
-    this.selectedVideoInputTransformDeviceObservers =
-      this.selectedVideoInputTransformDeviceObservers.filter(
-        (callback) => callback !== callbackToRemove
-      );
-  };
-
-  private publishSelectedVideoInputTransformDevice = (): void => {
-    for (const observer of this.selectedVideoInputTransformDeviceObservers) {
-      observer(this.selectedVideoInputTransformDevice);
-    }
-  };
-
   subscribeToSelectedAudioInputDevice = (
-    callback: (deviceId: string | null) => void
+    callback: (device: AudioInputDevice) => void
   ): void => {
     this.selectedAudioInputDeviceObservers.push(callback);
   };
 
-  subscribeToSelectedAudioInputTransformDevice = (
-    callback: (device: Device | AudioTransformDevice | null) => void
-  ): void => {
-    this.selectedAudioInputTransformDeviceObservers.push(callback);
-  };
-
   unsubscribeFromSelectedAudioInputDevice = (
-    callbackToRemove: (deviceId: string | null) => void
+    callbackToRemove: (device: AudioInputDevice) => void
   ): void => {
     this.selectedAudioInputDeviceObservers =
       this.selectedAudioInputDeviceObservers.filter(
         (callback) => callback !== callbackToRemove
       );
   };
 
-  unsubscribeFromSelectedAudioInputTransformDevice = (
-    callbackToRemove: (device: Device | AudioTransformDevice | null) => void
-  ): void => {
-    this.selectedAudioInputTransformDeviceObservers =
-      this.selectedAudioInputTransformDeviceObservers.filter(
-        (callback) => callback !== callbackToRemove
-      );
-  };
-
   private publishSelectedAudioInputDevice = (): void => {
     for (const observer of this.selectedAudioInputDeviceObservers) {
       observer(this.selectedAudioInputDevice);
     }
   };
 
-  private publishSelectedAudioInputTransformDevice = (): void => {
-    for (const observer of this.selectedAudioInputTransformDeviceObservers) {
-      observer(this.selectedAudioInputTransformDevice);
-    }
-  };
-
   subscribeToSelectedAudioOutputDevice = (
-    callback: (deviceId: string | null) => void
+    callback: (deviceId: string | undefined) => void
```

Comment:
In JS SDK, it is `string | null`. The default value for audio output device is `null` or default of browser. Don't think we have `undefined` value for output.

Reply:
`undefined` is not a value for any actual device selection. It's the value before component mount or after component unmount. Will change it to `deviceId: string | null | undefined`

Reply:
`undefined` does not make sense. No selection means `null` as we just use whatever the browser used. (i.e., if you do not make any explicit selection, the value is `null` and not `undefined`.

Reply:
Updated to `string | null` for `selectedAudioOutputDevice`